File: golang/cmd/badauth/badauth.go
// Usage: go run cmd/badauth/badauth.go
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/dnsimple/dnsimple-go/dnsimple"
	"golang.org/x/oauth2"
)

func main() {
	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: "bogus"})
	tc := oauth2.NewClient(context.Background(), ts)

	client := dnsimple.NewClient(tc)
	client.BaseURL = "https://api.sandbox.dnsimple.com"

	// get the current authenticated account (if you don't know who you are)
	whoamiResponse, err := client.Identity.Whoami(context.Background())
	if err != nil {
		fmt.Printf("Whoami() returned error: %v\n", err)
		os.Exit(1)
	}
	fmt.Println(whoamiResponse)
}

File: csharp/Examples/BadAuth.cs
using dnsimple;

namespace DnsimpleExamples.Examples;

// Usage: dotnet run --framework net10.0 -- bad_auth
//
// Note that this example intentionally sends a bogus access token to show
// what happens when authentication fails.
public static class BadAuth
{
    public static void Run(string[] args)
    {
        var client = new Client();
        client.ChangeBaseUrlTo("https://api.sandbox.dnsimple.com");
        client.AddCredentials(new OAuth2Credentials("bogus"));

        // get the current authenticated account (if you don't know who you are)
        try
        {
            var whoami = client.Identity.Whoami().Data;
            Console.WriteLine($"Account: Id={whoami.Account.Id}, Email={whoami.Account.Email}");
        }
        catch (AuthenticationException e)
        {
            // The API responds with a 401 for a bad token, which the client
            // surfaces as an AuthenticationException.
            Console.WriteLine($"Whoami() returned error: {e.Message}");
        }
    }
}
